<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>Form</class>
 <widget class="QWidget" name="Form">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>335</width>
    <height>741</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Form</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <layout class="QFormLayout" name="formLayout_2">
     <property name="labelAlignment">
      <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
     </property>
     <item row="0" column="0">
      <widget class="QLabel" name="label_12">
       <property name="font">
        <font>
         <weight>75</weight>
         <bold>true</bold>
        </font>
       </property>
       <property name="text">
        <string>Energy:</string>
       </property>
      </widget>
     </item>
     <item row="0" column="1">
      <layout class="QHBoxLayout" name="horizontalLayout_7">
       <property name="spacing">
        <number>2</number>
       </property>
       <item>
        <widget class="PyDMLabel" name="PyDMLabel_4">
         <property name="font">
          <font>
           <family>Monospace</family>
           <weight>75</weight>
           <bold>true</bold>
          </font>
         </property>
         <property name="toolTip">
          <string>The current readback value for the beamline energy, based on the monochromator setpoint.</string>
         </property>
         <property name="styleSheet">
          <string notr="true">font-weight: bold;</string>
         </property>
         <property name="text">
          <string>######â€‰eV</string>
         </property>
         <property name="showUnits" stdset="0">
          <bool>true</bool>
         </property>
         <property name="channel" stdset="0">
          <string>haven://${DEVICE}.energy.user_readback</string>
         </property>
        </widget>
       </item>
       <item>
        <spacer name="horizontalSpacer_5">
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>40</width>
           <height>20</height>
          </size>
         </property>
        </spacer>
       </item>
       <item>
        <widget class="PyDMRelatedDisplayButton" name="PyDMRelatedDisplayButton_4">
         <property name="enabled">
          <bool>true</bool>
         </property>
         <property name="sizePolicy">
          <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="minimumSize">
          <size>
           <width>0</width>
           <height>0</height>
          </size>
         </property>
         <property name="toolTip">
          <string/>
         </property>
         <property name="PyDMIcon" stdset="0">
          <string>cog</string>
         </property>
         <property name="filenames" stdset="0">
          <stringlist>
           <string>motor.py</string>
          </stringlist>
         </property>
         <property name="titles" stdset="0">
          <stringlist>
           <string>More</string>
          </stringlist>
         </property>
         <property name="showIcon" stdset="0">
          <bool>true</bool>
         </property>
         <property name="macros" stdset="0">
          <stringlist>
           <string>MOTOR=${DEVICE}.energy</string>
          </stringlist>
         </property>
         <property name="openInNewWindow" stdset="0">
          <bool>true</bool>
         </property>
        </widget>
       </item>
      </layout>
     </item>
     <item row="1" column="0">
      <widget class="QLabel" name="label_7">
       <property name="font">
        <font>
         <weight>50</weight>
         <bold>false</bold>
         <kerning>true</kerning>
        </font>
       </property>
       <property name="text">
        <string>Bragg:</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
       </property>
      </widget>
     </item>
     <item row="1" column="1">
      <layout class="QHBoxLayout" name="horizontalLayout_4">
       <item>
        <widget class="PyDMLabel" name="PyDMLabel_3">
         <property name="font">
          <font>
           <family>Monospace</family>
           <weight>50</weight>
           <bold>false</bold>
          </font>
         </property>
         <property name="toolTip">
          <string/>
         </property>
         <property name="text">
          <string>#####.###### arcsec</string>
         </property>
         <property name="precision" stdset="0">
          <number>0</number>
         </property>
         <property name="showUnits" stdset="0">
          <bool>true</bool>
         </property>
         <property name="precisionFromPV" stdset="0">
          <bool>true</bool>
         </property>
         <property name="channel" stdset="0">
          <string>haven://${DEVICE}.bragg.user_readback</string>
         </property>
        </widget>
       </item>
       <item>
        <spacer name="horizontalSpacer_4">
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>40</width>
           <height>20</height>
          </size>
         </property>
        </spacer>
       </item>
       <item>
        <widget class="PyDMRelatedDisplayButton" name="PyDMRelatedDisplayButton_3">
         <property name="enabled">
          <bool>true</bool>
         </property>
         <property name="sizePolicy">
          <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="minimumSize">
          <size>
           <width>0</width>
           <height>0</height>
          </size>
         </property>
         <property name="toolTip">
          <string/>
         </property>
         <property name="PyDMIcon" stdset="0">
          <string>cog</string>
         </property>
         <property name="filenames" stdset="0">
          <stringlist>
           <string>motor.py</string>
          </stringlist>
         </property>
         <property name="titles" stdset="0">
          <stringlist>
           <string>More</string>
          </stringlist>
         </property>
         <property name="showIcon" stdset="0">
          <bool>true</bool>
         </property>
         <property name="macros" stdset="0">
          <stringlist>
           <string>MOTOR=${DEVICE}.bragg</string>
          </stringlist>
         </property>
         <property name="openInNewWindow" stdset="0">
          <bool>true</bool>
         </property>
        </widget>
       </item>
      </layout>
     </item>
     <item row="2" column="0">
      <widget class="QLabel" name="label_15">
       <property name="text">
        <string>Offset:</string>
       </property>
      </widget>
     </item>
     <item row="2" column="1">
      <layout class="QHBoxLayout" name="horizontalLayout_5">
       <item>
        <widget class="PyDMLabel" name="PyDMLabel_8">
         <property name="font">
          <font>
           <family>Monospace</family>
          </font>
         </property>
         <property name="toolTip">
          <string/>
         </property>
         <property name="text">
          <string>####.###### um</string>
         </property>
         <property name="showUnits" stdset="0">
          <bool>true</bool>
         </property>
         <property name="channel" stdset="0">
          <string>haven://${DEVICE}.offset.user_readback</string>
         </property>
        </widget>
       </item>
       <item>
        <spacer name="horizontalSpacer_6">
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>40</width>
           <height>20</height>
          </size>
         </property>
        </spacer>
       </item>
       <item>
        <widget class="PyDMRelatedDisplayButton" name="PyDMRelatedDisplayButton_5">
         <property name="enabled">
          <bool>true</bool>
         </property>
         <property name="sizePolicy">
          <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="minimumSize">
          <size>
           <width>0</width>
           <height>0</height>
          </size>
         </property>
         <property name="toolTip">
          <string/>
         </property>
         <property name="PyDMIcon" stdset="0">
          <string>cog</string>
         </property>
         <property name="filenames" stdset="0">
          <stringlist>
           <string>motor.py</string>
          </stringlist>
         </property>
         <property name="titles" stdset="0">
          <stringlist>
           <string>More</string>
          </stringlist>
         </property>
         <property name="showIcon" stdset="0">
          <bool>true</bool>
         </property>
         <property name="macros" stdset="0">
          <stringlist>
           <string>MOTOR=${DEVICE}.offset</string>
          </stringlist>
         </property>
         <property name="openInNewWindow" stdset="0">
          <bool>true</bool>
         </property>
        </widget>
       </item>
      </layout>
     </item>
     <item row="3" column="0">
      <widget class="QLabel" name="label_16">
       <property name="text">
        <string>2nd Xtal Pitch:</string>
       </property>
      </widget>
     </item>
     <item row="3" column="1">
      <layout class="QHBoxLayout" name="horizontalLayout_6">
       <item>
        <widget class="PyDMLabel" name="PyDMLabel_6">
         <property name="font">
          <font>
           <family>Monospace</family>
          </font>
         </property>
         <property name="toolTip">
          <string/>
         </property>
         <property name="text">
          <string>#####.###### urad</string>
         </property>
         <property name="showUnits" stdset="0">
          <bool>true</bool>
         </property>
         <property name="channel" stdset="0">
          <string>haven://${DEVICE}.pitch2.user_readback</string>
         </property>
        </widget>
       </item>
       <item>
        <spacer name="horizontalSpacer_7">
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>40</width>
           <height>20</height>
          </size>
         </property>
        </spacer>
       </item>
       <item>
        <widget class="PyDMRelatedDisplayButton" name="PyDMRelatedDisplayButton_6">
         <property name="enabled">
          <bool>true</bool>
         </property>
         <property name="sizePolicy">
          <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="minimumSize">
          <size>
           <width>0</width>
           <height>0</height>
          </size>
         </property>
         <property name="toolTip">
          <string/>
         </property>
         <property name="PyDMIcon" stdset="0">
          <string>cog</string>
         </property>
         <property name="filenames" stdset="0">
          <stringlist>
           <string>motor.py</string>
          </stringlist>
         </property>
         <property name="titles" stdset="0">
          <stringlist>
           <string>More</string>
          </stringlist>
         </property>
         <property name="showIcon" stdset="0">
          <bool>true</bool>
         </property>
         <property name="macros" stdset="0">
          <stringlist>
           <string>MOTOR=${DEVICE}.pitch2</string>
          </stringlist>
         </property>
         <property name="openInNewWindow" stdset="0">
          <bool>true</bool>
         </property>
        </widget>
       </item>
      </layout>
     </item>
     <item row="4" column="0">
      <widget class="QLabel" name="label_17">
       <property name="text">
        <string>2nd Xtal Roll:</string>
       </property>
      </widget>
     </item>
     <item row="4" column="1">
      <layout class="QHBoxLayout" name="horizontalLayout_10">
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_8">
         <item>
          <widget class="PyDMLabel" name="PyDMLabel_7">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="font">
            <font>
             <family>Monospace</family>
            </font>
           </property>
           <property name="toolTip">
            <string/>
           </property>
           <property name="text">
            <string>#####.###### urad</string>
           </property>
           <property name="showUnits" stdset="0">
            <bool>true</bool>
           </property>
           <property name="channel" stdset="0">
            <string>haven://${DEVICE}.roll2.user_readback</string>
           </property>
          </widget>
         </item>
         <item>
          <spacer name="horizontalSpacer_8">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>40</width>
             <height>20</height>
            </size>
           </property>
          </spacer>
         </item>
         <item>
          <widget class="PyDMRelatedDisplayButton" name="PyDMRelatedDisplayButton_7">
           <property name="enabled">
            <bool>true</bool>
           </property>
           <property name="sizePolicy">
            <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="minimumSize">
            <size>
             <width>0</width>
             <height>0</height>
            </size>
           </property>
           <property name="toolTip">
            <string/>
           </property>
           <property name="PyDMIcon" stdset="0">
            <string>cog</string>
           </property>
           <property name="filenames" stdset="0">
            <stringlist>
             <string>motor.py</string>
            </stringlist>
           </property>
           <property name="titles" stdset="0">
            <stringlist>
             <string>More</string>
            </stringlist>
           </property>
           <property name="showIcon" stdset="0">
            <bool>true</bool>
           </property>
           <property name="macros" stdset="0">
            <stringlist>
             <string>MOTOR=${DEVICE}.roll2</string>
            </stringlist>
           </property>
           <property name="openInNewWindow" stdset="0">
            <bool>true</bool>
           </property>
          </widget>
         </item>
        </layout>
       </item>
      </layout>
     </item>
     <item row="5" column="0">
      <widget class="QLabel" name="label">
       <property name="text">
        <string>Mode:</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
       </property>
      </widget>
     </item>
     <item row="5" column="1">
      <widget class="PyDMEnumComboBox" name="PyDMEnumComboBox">
       <property name="toolTip">
        <string/>
       </property>
       <property name="channel" stdset="0">
        <string>haven://${DEVICE}.mode</string>
       </property>
      </widget>
     </item>
     <item row="6" column="0">
      <widget class="QLabel" name="label_8">
       <property name="text">
        <string>Feedback Î¸:</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
       </property>
      </widget>
     </item>
     <item row="6" column="1">
      <layout class="QHBoxLayout" name="horizontalLayout">
       <item>
        <widget class="PyDMByteIndicator" name="PyDMByteIndicator">
         <property name="enabled">
          <bool>false</bool>
         </property>
         <property name="sizePolicy">
          <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="toolTip">
          <string/>
         </property>
         <property name="offColor" stdset="0">
          <color>
           <red>255</red>
           <green>0</green>
           <blue>0</blue>
          </color>
         </property>
         <property name="showLabels" stdset="0">
          <bool>false</bool>
         </property>
        </widget>
       </item>
       <item>
        <widget class="PyDMPushButton" name="PyDMPushButton">
         <property name="enabled">
          <bool>false</bool>
         </property>
         <property name="toolTip">
          <string/>
         </property>
         <property name="text">
          <string>Disable</string>
         </property>
        </widget>
       </item>
      </layout>
     </item>
     <item row="7" column="0">
      <widget class="QLabel" name="label_9">
       <property name="text">
        <string>Feedback Ï‡:</string>
       </property>
      </widget>
     </item>
     <item row="7" column="1">
      <layout class="QHBoxLayout" name="horizontalLayout_2">
       <item>
        <widget class="PyDMByteIndicator" name="PyDMByteIndicator_2">
         <property name="enabled">
          <bool>false</bool>
         </property>
         <property name="sizePolicy">
          <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="toolTip">
          <string/>
         </property>
         <property name="offColor" stdset="0">
          <color>
           <red>255</red>
           <green>0</green>
           <blue>0</blue>
          </color>
         </property>
         <property name="showLabels" stdset="0">
          <bool>false</bool>
         </property>
        </widget>
       </item>
       <item>
        <widget class="PyDMPushButton" name="PyDMPushButton_2">
         <property name="enabled">
          <bool>false</bool>
         </property>
         <property name="toolTip">
          <string/>
         </property>
         <property name="text">
          <string>Disable</string>
         </property>
        </widget>
       </item>
      </layout>
     </item>
    </layout>
   </item>
   <item>
    <widget class="QGroupBox" name="groupBox">
     <property name="title">
      <string>Calibration</string>
     </property>
     <layout class="QFormLayout" name="formLayout">
      <property name="labelAlignment">
       <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
      </property>
      <item row="1" column="0">
       <widget class="QLabel" name="label_4">
        <property name="text">
         <string>Current Offset:</string>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="PyDMLabel" name="PyDMLabel">
        <property name="toolTip">
         <string/>
        </property>
        <property name="precision" stdset="0">
         <number>0</number>
        </property>
        <property name="showUnits" stdset="0">
         <bool>true</bool>
        </property>
        <property name="precisionFromPV" stdset="0">
         <bool>true</bool>
        </property>
        <property name="alarmSensitiveContent" stdset="0">
         <bool>false</bool>
        </property>
        <property name="alarmSensitiveBorder" stdset="0">
         <bool>true</bool>
        </property>
        <property name="PyDMToolTip" stdset="0">
         <string>The current value for the ${DEVICE} calibration offset.</string>
        </property>
        <property name="channel" stdset="0">
         <string>haven://${DEVICE}.transform_offset</string>
        </property>
        <property name="enableRichText" stdset="0">
         <bool>false</bool>
        </property>
       </widget>
      </item>
      <item row="2" column="0">
       <widget class="QLabel" name="label_2">
        <property name="text">
         <string>Truth:</string>
        </property>
       </widget>
      </item>
      <item row="2" column="1">
       <widget class="QDoubleSpinBox" name="truth_spinbox">
        <property name="suffix">
         <string>â€‰eV</string>
        </property>
        <property name="decimals">
         <number>1</number>
        </property>
        <property name="maximum">
         <double>100.000000000000000</double>
        </property>
       </widget>
      </item>
      <item row="3" column="0">
       <widget class="QLabel" name="label_3">
        <property name="text">
         <string>Dial:</string>
        </property>
       </widget>
      </item>
      <item row="3" column="1">
       <widget class="QDoubleSpinBox" name="dial_spinbox">
        <property name="suffix">
         <string>â€‰eV</string>
        </property>
        <property name="decimals">
         <number>1</number>
        </property>
        <property name="maximum">
         <double>100.000000000000000</double>
        </property>
       </widget>
      </item>
      <item row="4" column="1">
       <layout class="QHBoxLayout" name="horizontalLayout_3">
        <item>
         <widget class="QueueserverButton" name="calibrate_button">
          <property name="text">
           <string>Calibrate</string>
          </property>
         </widget>
        </item>
        <item>
         <spacer name="horizontalSpacer">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>40</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
       </layout>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QLabel" name="label_11">
     <property name="text">
      <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;img width=&quot;300&quot; src=&quot;:/optics/monochromator.png&quot;/&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>PyDMLabel</class>
   <extends>QLabel</extends>
   <header>pydm.widgets.label</header>
  </customwidget>
  <customwidget>
   <class>PyDMByteIndicator</class>
   <extends>QWidget</extends>
   <header>pydm.widgets.byte</header>
  </customwidget>
  <customwidget>
   <class>PyDMEnumComboBox</class>
   <extends>QComboBox</extends>
   <header>pydm.widgets.enum_combo_box</header>
  </customwidget>
  <customwidget>
   <class>PyDMPushButton</class>
   <extends>QPushButton</extends>
   <header>pydm.widgets.pushbutton</header>
  </customwidget>
  <customwidget>
   <class>PyDMRelatedDisplayButton</class>
   <extends>QPushButton</extends>
   <header>pydm.widgets.related_display_button</header>
  </customwidget>
  <customwidget>
   <class>QueueserverButton</class>
   <extends>QPushButton</extends>
   <header>firefly.queue_button</header>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="../resources/beamline_components.qrc"/>
 </resources>
 <connections/>
</ui>
